cinema/views.py

```diff
@@ -56,3 +84,61 @@
             return MovieSessionDetailSerializer
 
         return MovieSessionSerializer
+
+    def get_queryset(self):
+        queryset = self.queryset
+        date = self.request.query_params.get("date")
+        movie = self.request.query_params.get("movie")
+
+        if date:
+            try:
+                date = datetime.strptime(date, "%Y-%m-%d").date()
+                queryset = queryset.filter(show_time__date=date)
+            except ValueError:
+                pass
+
+        if movie:
+            queryset = queryset.filter(movie__id=movie)
+
+        if self.action in "list":
+            queryset = (
+                queryset
+                .select_related()
+                .annotate(tickets_available=F(
+                    "cinema_hall__seats_in_row") * F(
+                    "cinema_hall__rows") - Count("tickets"))
+            )
+        elif self.action == "retrieve":
+            queryset = queryset.select_related("movie")
+        return queryset.distinct()
+
+
+class OrderSetPagination(PageNumberPagination):
+    page_size = 1
+    page_size_query_param = "page_size"
+    max_page_size = 20
+
+
+class OrderViewSet(viewsets.ModelViewSet):
+    queryset = Order.objects.all()
+    serializer_class = OrderSerializer
+    pagination_class = OrderSetPagination
+
+    def get_queryset(self):
+        queryset = self.queryset.filter(user=self.request.user)
+
+        if self.action in "list":
+            queryset = queryset.prefetch_related(
+                "tickets__movie_session__movie")
+        return queryset
+
+    def perform_create(self, serializer):
+        serializer.save(user=self.request.user)
+
+    def get_serializer_class(self):
+        serializer = self.serializer_class
+
+        if self.action in "list":
+            serializer = OrderListSerializer
```

Comment:
The condition `if self.action in 'list':` should be `if self.action == 'list':` to correctly check if the action is 'list'.